lib/dao.js

```diff
@@ -3059,7 +3059,10 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
 
         function replaceCallback(err, data) {
           if (err) return cb(err);
-
+          // if connector is cloudant, add the `_rev` property back before passing it to the user
+          if (connector.name === 'cloudant') {
+            context.data._rev = data._rev;
```

Comment:
I do not agree. If the user wants to replace {"name": "a", "age": 1} where `id : 3`, then the database will return back that `OK` I was able to replace the instance with the data you gave me and return the new data which should be the same as the data user initially passed.

In cloudant's case, we just need to replace the new `_rev` back or else the user gets back the old `_rev` value.